tests/php/Security/RememberLoginHashTest.php

```diff
@@ -95,4 +95,32 @@ public function testGetSetLogoutAcrossDevices()
         RememberLoginHash::setLogoutAcrossDevices(false);
         $this->assertFalse(RememberLoginHash::getLogoutAcrossDevices());
     }
+
+    public function testRenew()
+    {
+        // If session-manager module is installed it expects an active request during renewal
+        if (class_exists(LoginSession::class)) {
+            $this->markTestSkipped();
+        }
+
+        $member = $this->objFromFixture(Member::class, 'main');
+
+        // With token replacement enabled, hash and token should change
+        RememberLoginHash::config()->set('replace_token_during_session_renewal', true);
+        $hash = RememberLoginHash::generate($member);
+        $oldToken = $hash->getToken();
+        $oldHash = $hash->Hash;
+        $hash->renew();
+        $this->assertNotEquals($oldToken, $hash->getToken());
+        $this->assertNotEquals($oldHash, $hash->Hash);
+
+        // With token regeneration disabled, hash should remain the same and token should remain empty
+        RememberLoginHash::config()->set('replace_token_during_session_renewal', false);
+        $hash = RememberLoginHash::generate($member);
+        $hash->setToken('');
+        $oldHash = $hash->Hash;
+        $hash->renew();
+        $this->assertEmpty($hash->getToken());
+        $this->assertEquals($oldHash, $hash->Hash);
```

Comment:
Since the code here is basically identical, turn this into a [`@dataProvider`](https://docs.phpunit.de/en/9.6/writing-tests-for-phpunit.html#data-providers) so we just have the code once.

```suggestion
        RememberLoginHash::config()->set('replace_token_during_session_renewal', $replaceToken);
        $hash = RememberLoginHash::generate($member);
        $oldToken = $hash->getToken();
        $oldHash = $hash->Hash;
        Deprecation::withNoReplacement(fn() => $hash->renew());
        if ($replaceToken) {
            $this->assertNotEquals($oldToken, $hash->getToken());
            $this->assertNotEquals($oldHash, $hash->Hash);
        } else {
            $this->assertEmpty($hash->getToken());
            $this->assertEquals($oldHash, $hash->Hash);
        }
```
```php
public function provideRenew(): array
{
    return [
        [true],
        [false],
    ];
}
```